lighthouse-extension/gulpfile.js

```diff
@@ -108,7 +108,13 @@ gulp.task('browserify-lighthouse', () => {
     'app/src/lighthouse-ext-background.js',
   ], {read: false})
     .pipe(tap(file => {
-      let bundle = browserify(file.path); // , {debug: true}); // for sourcemaps
+      const environment = file.path.includes('-ext-') ? 'Extension' : 'DevTools';
+      let bundle = browserify(file.path, {
+        insertGlobalVars: {
+          'global.ENVIRONMENT': () => `"${environment}"`,
```

Comment:
I won't be able to answer any questions starting with "why" regarding this line. It's just how it's done in browserify e.g. we can't just have `() => environment` here because we will end up with `{"ENVIRONMENT": DevTools}` (note missing quotes) in the result code 🤷‍♂️

Reply:
sticking with insertGlobalVars is in my opinion a better ideas as it's native browserify. We can still use env vars with `process.env` which makes it a bit more verbose. I would try to use JSON.stringify as it's more common due webpacks implementation as patrick suggested.